pkg/prometheus/promcfg.go

```diff
@@ -322,12 +322,32 @@ func honorLabels(userHonorLabels, overrideHonorLabels bool) bool {
 	return userHonorLabels
 }
 
-func (cg *configGenerator) generatePodMonitorConfig(version semver.Version,
+// honorTimestamps adds option to enforce honor_timestamps option in scrape_config.
+// We want to disable honoring timestamps when user specified it or when global
+// override is set. For backwards compatibility with prometheus <2.9.0 we don't
```

Comment:
Nevermind, it looks like we're doing that now outside of this function.

Reply:
Added version protection in each `generate*MonitorConfig`. I didn't want to do it in honorTimestamps() as all version protections are in `generate*MonitorConfig` functions and it might cloud codebase a bit.